gobblin-core/src/main/java/gobblin/writer/PartitionedDataWriter.java

```diff
@@ -171,4 +172,24 @@ public void close() throws IOException {
     return this.builder.get().forPartition(partition).withWriterId(this.baseWriterId + "_" + this.writerIdSuffix++)
         .build();
   }
+
+  @Override
+  public State getFinalState() {
+
+    State state = new State();
+    try {
+      for (Map.Entry<GenericRecord, DataWriter<D>> entry : this.partitionWriters.asMap().entrySet()) {
+        if (entry.getValue() instanceof FinalState) {
+          state.addAll(((FinalState)entry.getValue()).getFinalState());
```

Comment:
This will overwrite values that are repeated in multiple writers. Can we keep them separate somehow? Like adding partition info as suffix?